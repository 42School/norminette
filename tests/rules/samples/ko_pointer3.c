int	main(void) {
	int	***	*	a;
	int			b;	if 					(5 * 		 *** a)
		b = 		 	12;
}

int	main(void) {
	int	***	*	a;
	int			b;	if 					(5 * 		 ***a)
		b = 		 	12;
}

int	main(void) {
	int	***	*	a;
	int			b;	a = b * (*(*(*(*a) ) ) );
	(*a 	 					) = b * 				a;
}

int	main(void) {
	int	toto tata
	*	 * * *	a;
	int			b;
}