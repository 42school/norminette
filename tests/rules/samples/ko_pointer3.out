[36mko_pointer3.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 1":
		<INT> <TAB> <IDENTIFIER=main> <LPARENTHESIS> <VOID> <RPARENTHESIS> <SPACE> 
[36mko_pointer3.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 1":
		<LBRACE> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 2":
		<TAB> <INT> <TAB> <MULT> <MULT> <MULT> <TAB> <MULT> <TAB> <IDENTIFIER=a> <SEMI_COLON> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 3":
		<TAB> <INT> <TAB> <TAB> <TAB> <IDENTIFIER=b> <SEMI_COLON> <TAB> 
[36mko_pointer3.c[0m - [32mIsControlStatement[0m In "Function" from "GlobalScope" line 3":
		<IF> <SPACE> <TAB> <TAB> <TAB> <TAB> <TAB> <LPARENTHESIS> <CONSTANT=5> <SPACE> <MULT> <SPACE> <TAB> <TAB> <SPACE> <MULT> <MULT> <MULT> <SPACE> <IDENTIFIER=a> <RPARENTHESIS> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsAssignation[0m In "ControlStructure" from "Function" line 4":
		<TAB> <TAB> <IDENTIFIER=b> <SPACE> <ASSIGN> <SPACE> <TAB> <TAB> <SPACE> <TAB> <CONSTANT=12> <SEMI_COLON> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 5":
		<RBRACE> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 6":
		<NEWLINE>
[36mko_pointer3.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 7":
		<INT> <TAB> <IDENTIFIER=main> <LPARENTHESIS> <VOID> <RPARENTHESIS> <SPACE> 
[36mko_pointer3.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 7":
		<LBRACE> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 8":
		<TAB> <INT> <TAB> <MULT> <MULT> <MULT> <TAB> <MULT> <TAB> <IDENTIFIER=a> <SEMI_COLON> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 9":
		<TAB> <INT> <TAB> <TAB> <TAB> <IDENTIFIER=b> <SEMI_COLON> <TAB> 
[36mko_pointer3.c[0m - [32mIsControlStatement[0m In "Function" from "GlobalScope" line 9":
		<IF> <SPACE> <TAB> <TAB> <TAB> <TAB> <TAB> <LPARENTHESIS> <CONSTANT=5> <SPACE> <MULT> <SPACE> <TAB> <TAB> <SPACE> <MULT> <MULT> <MULT> <IDENTIFIER=a> <RPARENTHESIS> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsAssignation[0m In "ControlStructure" from "Function" line 10":
		<TAB> <TAB> <IDENTIFIER=b> <SPACE> <ASSIGN> <SPACE> <TAB> <TAB> <SPACE> <TAB> <CONSTANT=12> <SEMI_COLON> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 11":
		<RBRACE> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 12":
		<NEWLINE>
[36mko_pointer3.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 13":
		<INT> <TAB> <IDENTIFIER=main> <LPARENTHESIS> <VOID> <RPARENTHESIS> <SPACE> 
[36mko_pointer3.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 13":
		<LBRACE> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 14":
		<TAB> <INT> <TAB> <MULT> <MULT> <MULT> <TAB> <MULT> <TAB> <IDENTIFIER=a> <SEMI_COLON> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 15":
		<TAB> <INT> <TAB> <TAB> <TAB> <IDENTIFIER=b> <SEMI_COLON> <TAB> 
[36mko_pointer3.c[0m - [32mIsAssignation[0m In "Function" from "GlobalScope" line 15":
		<IDENTIFIER=a> <SPACE> <ASSIGN> <SPACE> <IDENTIFIER=b> <SPACE> <MULT> <SPACE> <LPARENTHESIS> <MULT> <LPARENTHESIS> <MULT> <LPARENTHESIS> <MULT> <LPARENTHESIS> <MULT> <IDENTIFIER=a> <RPARENTHESIS> <SPACE> <RPARENTHESIS> <SPACE> <RPARENTHESIS> <SPACE> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsAssignation[0m In "Function" from "GlobalScope" line 16":
		<TAB> <LPARENTHESIS> <MULT> <IDENTIFIER=a> <SPACE> <TAB> <SPACE> <TAB> <TAB> <TAB> <TAB> <TAB> <RPARENTHESIS> <SPACE> <ASSIGN> <SPACE> <IDENTIFIER=b> <SPACE> <MULT> <SPACE> <TAB> <TAB> <TAB> <TAB> <IDENTIFIER=a> <SEMI_COLON> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 17":
		<RBRACE> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 18":
		<NEWLINE>
[36mko_pointer3.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 19":
		<INT> <TAB> <IDENTIFIER=main> <LPARENTHESIS> <VOID> <RPARENTHESIS> <SPACE> 
[36mko_pointer3.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 19":
		<LBRACE> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 20":
		<TAB> <INT> <TAB> <IDENTIFIER=toto> <SPACE> <IDENTIFIER=tata> <NEWLINE>
		<TAB> <MULT> <TAB> <SPACE> <MULT> <SPACE> <MULT> <SPACE> <MULT> <TAB> <IDENTIFIER=a> <SEMI_COLON> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsVarDeclaration[0m In "Function" from "GlobalScope" line 22":
		<TAB> <INT> <TAB> <TAB> <TAB> <IDENTIFIER=b> <SEMI_COLON> <NEWLINE>
[36mko_pointer3.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 23":
		<RBRACE> 
ko_pointer3.c: Error!
Error: INVALID_HEADER       (line:   1, col:   1):	[97mMissing or invalid 42 header[0m
Error: BRACE_NEWLINE        (line:   1, col:  16):	[93mExpected newline before brace[0m
Error: SPC_AFTER_POINTER    (line:   2, col:  11):	[94mspace after pointer[0m
Error: SPC_AFTER_POINTER    (line:   2, col:  13):	[94mspace after pointer[0m
Error: NL_AFTER_VAR_DECL    (line:   3, col:  21):	[0mVariable declarations must be followed by a newline[0m
Error: TOO_FEW_TAB          (line:   3, col:  21):	[0mMissing tabs for indent level[0m
Error: TOO_MANY_INSTR       (line:   3, col:  21):	[0mToo many instructions on a single line[0m
Error: MIXED_SPACE_TAB      (line:   3, col:  23):	[93mMixed spaces and tabs[0m
Error: TAB_INSTEAD_SPC      (line:   3, col:  24):	[94mFound tab when expecting space[0m
Error: SPC_AFTER_POINTER    (line:   3, col:  56):	[94mspace after pointer[0m
Error: SPC_AFTER_POINTER    (line:   3, col:  57):	[94mspace after pointer[0m
Error: MIXED_SPACE_TAB      (line:   4, col:  12):	[93mMixed spaces and tabs[0m
Error: TAB_INSTEAD_SPC      (line:   4, col:  13):	[94mFound tab when expecting space[0m
Error: BRACE_NEWLINE        (line:   7, col:  16):	[93mExpected newline before brace[0m
Error: SPC_AFTER_POINTER    (line:   8, col:  11):	[94mspace after pointer[0m
Error: SPC_AFTER_POINTER    (line:   8, col:  13):	[94mspace after pointer[0m
Error: NL_AFTER_VAR_DECL    (line:   9, col:  21):	[0mVariable declarations must be followed by a newline[0m
Error: TOO_FEW_TAB          (line:   9, col:  21):	[0mMissing tabs for indent level[0m
Error: TOO_MANY_INSTR       (line:   9, col:  21):	[0mToo many instructions on a single line[0m
Error: MIXED_SPACE_TAB      (line:   9, col:  23):	[93mMixed spaces and tabs[0m
Error: TAB_INSTEAD_SPC      (line:   9, col:  24):	[94mFound tab when expecting space[0m
Error: MIXED_SPACE_TAB      (line:  10, col:  12):	[93mMixed spaces and tabs[0m
Error: TAB_INSTEAD_SPC      (line:  10, col:  13):	[94mFound tab when expecting space[0m
Error: BRACE_NEWLINE        (line:  13, col:  16):	[93mExpected newline before brace[0m
Error: SPC_AFTER_POINTER    (line:  14, col:  11):	[94mspace after pointer[0m
Error: SPC_AFTER_POINTER    (line:  14, col:  13):	[94mspace after pointer[0m
Error: NL_AFTER_VAR_DECL    (line:  15, col:  21):	[0mVariable declarations must be followed by a newline[0m
Error: TOO_FEW_TAB          (line:  15, col:  21):	[0mMissing tabs for indent level[0m
Error: TOO_MANY_INSTR       (line:  15, col:  21):	[0mToo many instructions on a single line[0m
Error: NO_SPC_AFR_PAR       (line:  15, col:  38):	[94mExtra space after parenthesis (brace/bracket)[0m
Error: NO_SPC_AFR_PAR       (line:  15, col:  40):	[94mExtra space after parenthesis (brace/bracket)[0m
Error: NO_SPC_BFR_PAR       (line:  15, col:  40):	[94mExtra space before parenthesis (brace/bracket)[0m
Error: NO_SPC_AFR_PAR       (line:  15, col:  42):	[94mExtra space after parenthesis (brace/bracket)[0m
Error: NO_SPC_BFR_PAR       (line:  15, col:  42):	[94mExtra space before parenthesis (brace/bracket)[0m
Error: NO_SPC_BFR_PAR       (line:  15, col:  44):	[94mExtra space before parenthesis (brace/bracket)[0m
Error: MIXED_SPACE_TAB      (line:  16, col:   8):	[93mMixed spaces and tabs[0m
Error: TAB_INSTEAD_SPC      (line:  16, col:   9):	[94mFound tab when expecting space[0m
Error: NO_SPC_BFR_PAR       (line:  16, col:  33):	[94mExtra space before parenthesis (brace/bracket)[0m
Error: BRACE_NEWLINE        (line:  19, col:  16):	[93mExpected newline before brace[0m
Error: TAB_REPLACE_SPACE    (line:  20, col:   8):	[0mFound tab when expecting space[0m
Error: SPACE_REPLACE_TAB    (line:  20, col:  13):	[0mFound space when expecting tab[0m
Error: TOO_FEW_TAB          (line:  21, col:   1):	[0mMissing tabs for indent level[0m
Error: SPC_AFTER_POINTER    (line:  21, col:   5):	[94mspace after pointer[0m
Error: MIXED_SPACE_TAB      (line:  21, col:   6):	[93mMixed spaces and tabs[0m
Error: SPC_AFTER_POINTER    (line:  21, col:  10):	[94mspace after pointer[0m
Error: SPC_AFTER_POINTER    (line:  21, col:  12):	[94mspace after pointer[0m
Error: SPC_AFTER_POINTER    (line:  21, col:  14):	[94mspace after pointer[0m
Error: BRACE_SHOULD_EOL     (line:  23, col:   1):	[0mExpected newline after brace[0m
